x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/TransportFreezeIndexAction.java

```diff
@@ -108,6 +114,32 @@ protected void masterOperation(FreezeRequest request, ClusterState state, Action
             listener.onResponse(new FreezeResponse(true, true));
             return;
         }
+
+        final CloseIndexClusterStateUpdateRequest closeRequest = new CloseIndexClusterStateUpdateRequest()
+            .ackTimeout(request.timeout())
+            .masterNodeTimeout(request.masterNodeTimeout())
+            .indices(concreteIndices);
+
+        indexStateService.closeIndices(closeRequest, new ActionListener<AcknowledgedResponse>() {
+            @Override
+            public void onResponse(final AcknowledgedResponse response) {
+                if (response.isAcknowledged()) {
+                    toggleFrozenSettings(concreteIndices, request, listener);
+                } else {
+                    listener.onResponse(new FreezeResponse(false, false));
```

Comment:
add TODO here that we should also extend the FreezeResponse object with info about failed closing?